src/app/shared/components/product/product-quantity/product-quantity.component.ts

```diff
@@ -70,4 +70,20 @@ export class ProductQuantityComponent implements OnInit {
   change(target: EventTarget) {
     this.setValue(Number.parseInt((target as HTMLDataElement).value, 10));
   }
+
+  input(event: Event) {
+    const inputEvent = event as InputEvent;
+    const inputData = inputEvent.data;
+    const inputField = inputEvent.target as HTMLInputElement;
+    const allowed = /^[^0-9]*$/;
+    if (inputData !== null && allowed.test(inputData)) {
+      inputField.value = `${this.context.get('quantity')}`;
+    }
+  }
+
+  paste(event: Event) {
+    const clipboardData = (event as ClipboardEvent).clipboardData.getData('text');
+    const allowed = /^[0-9]*$/;
+    return allowed.test(clipboardData);
+  }
```

Comment:
Instead of interrupting the `input` and `paste`, I think this can be done in the `product-context.facade.ts` by adding another validation, that the number must be positive:
https://github.com/intershop/intershop-pwa/blob/932473360d69774c2fa90649736e7638f1412469/src/app/core/facades/product-context.facade.ts#L191-L199

That way it will also become a fix for #849 again 😃

Reply:
If the number is negative, it is also below the minOrderQuantity. So negative numbers already trigger a warning. Problem with #849 is, that invalid numbers only show a warning but don't disable the checkout button. This PR was just an idea to solve this issue by not letting the user even enter invalid values. But this solution only prevents invalid characters not invalid quanities. And preventing the user from also entering invalid (positive) quantities would affect the user experience maybe a bit too much. Thinking about that, this whole PR might be kind of pointless as it doesn't really fix anything or bring some kind of actual use. So in the end sorry for wasting time, I guess?

Reply:
Oh I see, yes. Then it won't be a fix for the issue. But I think adding a pattern and preventing the paste is also a good improvement (and definitely not a waste of time :P).

I think we should take it like that. @shauke agreed?

Reply:
The changes of this PR can definitely not be a fix for #849. The problem there is that the checkout button is somehow not connected to the invalid state of the form. So preventing invalid input will somewhat help with this issue but not fix it for all relevant problems, e.g. a missing cost center selection. This needs to be fixed elsewhere.
Never the less this PR is not a waste of time since we all learned something and the input pattern definitely improves something. But I removed the input and paste handling. Chromium based browsers will already handle this pretty well on there own and prevent invalid inputs based on the provided patterns and in any other case a validation error message is provided. And if someone really wants to paste invalid content at this place in Firefox or Safari why shouldn't we let them? But it is good to know how we could prevent such things if it would matter. ;)